### PR TITLE
Features/exams overview

### PR DESCRIPTION
**Pull request by miljevictamara**

### CODE REVIEW

File: Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    public class Response
+    {
+        public IEnumerable<ResponseItem> Items { get; set; } = [];
+    }
+    public class ResponseItem
+    {
+        public long Id { get; set; }
+        public DateTime StartTime { get; set; }
+        public DateTime? EndTime { get; set; }
```
Comment (Batranovic) [inline Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs:20]: Are you using this anywhere? If not, maybe you should remove it.

File: Konteh/back-office-app/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -4,15 +4,18 @@ import { QuestionsOverviewComponent } from './features/questions/questions-overv
 import { HomeComponent } from './features/home/home.component';
 import { CreateQuestionComponent } from './features/questions/create-question/create-question.component';
 import { AuthGuard } from './authorization/auth.guard';
+import { ExamOverviewComponent } from './features/exams/exam-overview/exam-overview.component';
```
Comment (Batranovic) [inline Konteh/back-office-app/src/app/app-routing.module.ts:7]: maybe rename this component

**Review by Batranovic — COMMENTED** [review]
consider my review

File: Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+public static class GetAllExams
+{
+    public class Query : IRequest<Response>;
+
+    public class Response
+    {
+        public IEnumerable<ResponseItem> Items { get; set; } = [];
+    }
+    public class ResponseItem
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs:16]: You don't need an extra class, just make it IRequest<IEnumerable<Response>>

File: Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+        public async Task<Response> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var exams = await _repository.GetAll();
+            return new Response
+            {
+                Items = exams.Select(e => new ResponseItem
+                {
+                    Id = e.Id,
+                    Candidate = $"{e.Candiate.Name} {e.Candiate.Surname}, {e.Candiate.Faculty}",
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Features/Exams/GetAllExams.cs:41]: Name + Surname is enough, we don't want too much info in the table.

File: Konteh/Konteh.Domain/Enumerations/ExamStatus.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Konteh.Domain.Enumerations;
+
+public enum ExamStatus
+{
+    Pending = 1,
```
Comment (laslorajda) [inline Konteh/Konteh.Domain/Enumerations/ExamStatus.cs:5]: In hindsight, maybe "pending" doesn't make much sense.

File: Konteh/Konteh.Infrastructure/Migrations/20241025115719_AfterMerge.Designer.cs
Comment (laslorajda) [inline Konteh/Konteh.Infrastructure/Migrations/20241025115719_AfterMerge.Designer.cs:1]: This is very very scary. We need to consolidate migrations early next week.

File: Konteh/Konteh.Infrastructure/Repositories/ExamRepository.cs [hunk 6]
```diff
@@ -7,4 +8,12 @@ public class ExamRepository : BaseRepository<Exam>
     public ExamRepository(AppDbContext context) : base(context)
     {
     }
+
+    public override async Task<IEnumerable<Exam>> GetAll()
```
Comment (laslorajda) [inline Konteh/Konteh.Infrastructure/Repositories/ExamRepository.cs:12]: There we go! Excellent!

File: Konteh/back-office-app/src/app/app-routing.module.ts [hunk 7]
```diff
@@ -5,14 +5,17 @@ import { HomeComponent } from './features/home/home.component';
...
+import { ExamsOverviewComponent } from './features/exams/exams-overview/exams-overview.component';
 
 const routes: Routes = [
   { path: '', component: HomeComponent},
   { path: 'questions-overview', component: QuestionsOverviewComponent, canActivate:[AuthGuard]},
   { path: 'exam-notifications', component: ExamNotificationsComponent, canActivate:[AuthGuard]},
   { path: 'create-question', component: CreateQuestionComponent, canActivate: [AuthGuard] },
-  { path: 'edit-question/:id', component: CreateQuestionComponent, canActivate: [AuthGuard] }
+  { path: 'edit-question/:id', component: CreateQuestionComponent, canActivate: [AuthGuard] },
+  { path: 'exams-overview', component: ExamsOverviewComponent, canActivate: [AuthGuard] }
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/app-routing.module.ts:16]: We need to do something about the routing, please remind me next week.

File: Konteh/back-office-app/src/app/features/exams/exams-overview/exams-overview.component.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<div class="exam-table-container">
+    <div class="mat-elevation-z8">
+      <table mat-table [dataSource]="dataSource" class="mat-elevation-z8">
+  
+        <ng-container matColumnDef="id">
+          <th mat-header-cell *matHeaderCellDef> ID </th>
+          <td mat-cell *matCellDef="let exam"> {{ exam.id }} </td>
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/features/exams/exams-overview/exams-overview.component.html:7]: I don't see why we would ever display Ids in tables.

File: Konteh/back-office-app/src/app/features/exams/exams-overview/exams-overview.component.html [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<div class="exam-table-container">
+    <div class="mat-elevation-z8">
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/features/exams/exams-overview/exams-overview.component.html:2]: Maybe add a search by candidate name?

File: Konteh/back-office-app/src/app/features/layout/nav-bar/nav-bar.component.html [hunk 10]
```diff
@@ -1,17 +1,19 @@
 <mat-toolbar color="primary">
-      <span>Back Office</span>
+  
+  <img src="/logo.png" alt="image" class="logo">
       <span class="menu-spacer"></span>
       <a mat-button (click)="redirectToHome()">Home</a>
       <a mat-button (click)="redirectToQuestionsOverview()">Questions Overview</a>
       <a mat-button (click)="redirectToExamOverview()">Exams</a>
       <a mat-button (click)="redirectToCreateQuestion()">Create Question</a>
+      <a mat-button (click)="redirectToExamsOverview()">Exams Overview</a>
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/features/layout/nav-bar/nav-bar.component.html:9]: https://angular.dev/tutorials/learn-angular/14-routerLink

**Review by laslorajda — COMMENTED** [review]
A little cleaup needed

**Pull request merged**